### PR TITLE
Add environment creation to jujulib.

### PR DESCRIPTION
**Pull request by jaycee**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3212/
Test PASSed.

**Comment by jaycee** [comment]
Edit `development.ini` and set `jujugui.jem_url` to http://54.205.45.221:8082

Start the gui and open it; also open the browser development console. Confirm the gui starts up normally and that the "environment listing success" message and data is printed in the console. Play with the GUI a bit to make sure it's working normally.

In the console run `app.jem.newEnvironment('$your-user-name', '$your-environment-name', 'jcsackett/base', 'password', function(xhr) { console.log('success') }, function(xhr) {console.log(xhr.target.responseText) })` substituting in your user name and an env name. Make sure you see the success message.

Run `app.jem.listEnvironments(function(data) {console.log(data)})`; confirm you see your environment in the array of environment objects.

**Comment by kadams54** [comment]
:+1: LGTM.

**Comment by bac** [comment]
Code :+1: and I'm doing QA now.

File: jujugui/static/gui/src/app/assets/javascripts/jujulib/juju.js [hunk 1]
```diff
@@ -61,32 +70,76 @@ var module = module;
...
+  environment.prototype.newEnvironment = function (
+      envOwnerName, envName, baseTemplate, password, success, failure) {
+    var body = {
+      name: envName,
+      password: password,
+      templates: [baseTemplate]
+    };
+    var url = [this.jemUrl, 'env', envOwnerName].join('/');
+    var _newEnvironment = function(servers) {
+      var path = servers[0].path;
```
Comment (urosj) [inline jujugui/static/gui/src/app/assets/javascripts/jujulib/juju.js]: What if servers[] is empty?

Comment (jaycee) [inline jujugui/static/gui/src/app/assets/javascripts/jujulib/juju.js]: Servers should never be empty, but I'll add error handling.

**Comment by jaycee** [comment]
Update to QA instructions:

When you update `development.ini` to set `jujugui.jem_url`, also set `jujugui.interactive_login` to `true`.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3213/
Test PASSed.

**Comment by bac** [comment]
Attempted to QA with assistance by @jaycee .  Templates were not accessible so it all failed.  Marking as QA-Impossiblé.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3237/
Test PASSed.

**Comment by jaycee** [comment]
@urosj was able to successfully qa this. :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**